telematic_system/historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/service_manager.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (C) 2024 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+import asyncio
+from nats.aio.client import Client as NATS
+import os
+import logging
+from logging.handlers import RotatingFileHandler
+from datetime import datetime
+from enum import Enum
+from .rosbag_processor import Rosbag2Parser
+
+class LogType(Enum):
+    FILE = "file"
+    CONSOLE = "console"
+    ALL = "all"
+
+class ServiceManager:
+
+    rosbag_queue = []
+    def __init__(self):
+        #NATS client to receive requests from the web ui
+        self.nc = NATS()
+        # List of rosbags to be processeds
+        self.rosbag_queue = ['rosbag2_2024_01_26-15_00_24_0.mcap']
```

Comment:
need to read from the NATS topic, refer to the readme: https://github.com/usdot-fhwa-stol/cda-telematics/pull/174

Reply:
Yes, this functionality needs to be added. Currently hardcoded for testing